src/main/java/net/glowstone/generator/NetherGenerator.java

```diff
@@ -0,0 +1,288 @@
+package net.glowstone.generator;
+
+import java.util.Map;
+import java.util.Random;
+
+import net.glowstone.generator.populators.NetherPopulator;
+import net.glowstone.util.noise.PerlinOctaveGenerator;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.util.noise.OctaveGenerator;
+
+public class NetherGenerator extends GlowChunkGenerator {
+
+    private static final double COORDINATE_SCALE = 684.412D;   // coordinateScale
+    private static final double HEIGHT_SCALE = COORDINATE_SCALE * 3.0D; // heightScale
+    private static final double HEIGHT_NOISE_SCALE_X = 100.0D; // depthNoiseScaleX
+    private static final double HEIGHT_NOISE_SCALE_Z = 100.0D; // depthNoiseScaleZ
+    private static final double DETAIL_NOISE_SCALE_X = 80.0D;  // mainNoiseScaleX
+    private static final double DETAIL_NOISE_SCALE_Y = 60.0D;  // mainNoiseScaleY
+    private static final double DETAIL_NOISE_SCALE_Z = 80.0D;  // mainNoiseScaleZ
+    private static final double SURFACE_SCALE = 0.0625D;
+
+    private final double[] density = new double[5 * 17 * 5];
+
+    public NetherGenerator() {
+        super(new NetherPopulator());
+    }
+
+    @Override
+    public short[][] generateExtBlockSectionsWithData(World world, Random random, int chunkX, int chunkZ, BiomeGrid biomes) {
+        final short[][] buf = generateRawTerrain(world, chunkX, chunkZ);
+
+        int cx = chunkX << 4;
+        int cz = chunkZ << 4;
+
+        final double[] surfaceNoise = ((PerlinOctaveGenerator) getWorldOctaves(world).get("surface")).fBm(cx, cz, 0, 0.5D, 2.0D);
+        final double[] soulsandNoise = ((PerlinOctaveGenerator) getWorldOctaves(world).get("soulsand")).fBm(cx, cz, 0, 0.5D, 2.0D);
+        final double[] gravelNoise = ((PerlinOctaveGenerator) getWorldOctaves(world).get("gravel")).fBm(cx, 0, cz, 0.5D, 2.0D);
```

Comment:
This seems to be generating the gravel, though you probably knew that already :)